lecture/build.gradle

```
@@ -0,0 +1,289 @@
+    implementation "org.springframework.boot:spring-boot-starter-data-jpa:${spring_boot_version}"
+    implementation("org.springframework.boot:spring-boot-starter-web:${spring_boot_version}") {
+        exclude module: "spring-boot-starter-undertow"
+    }
+    implementation "org.springframework.boot:spring-boot-starter-tomcat:${spring_boot_version}"
+    implementation "org.springframework.boot:spring-boot-starter-websocket:${spring_boot_version}"
+    implementation "org.springframework.boot:spring-boot-starter-undertow:${spring_boot_version}"
+
+    implementation "org.springframework.ldap:spring-ldap-core:${spring_ldap_core_version}"
+    implementation "org.springframework.data:spring-data-ldap:${spring_data_ldap_version}"
+
+    implementation "com.fasterxml.jackson.module:jackson-module-jaxb-annotations"
+    implementation "javax.annotation:javax.annotation-api"
+    implementation "io.dropwizard.metrics:metrics-core"
+
+    implementation "org.springframework.cloud:spring-cloud-starter"
+    implementation "org.springframework.cloud:spring-cloud-starter-netflix-hystrix"
+    implementation "org.springframework.retry:spring-retry"
+    implementation "org.springframework.cloud:spring-cloud-starter-bootstrap"
+    implementation "org.springframework.cloud:spring-cloud-starter-openfeign"
+    implementation "org.springframework.security:spring-security-data:${spring_security_version}"
+    implementation "org.xmlbeam:xmlprojector:${xmlbeam_xmlprojector_version}"
+    implementation "mysql:mysql-connector-java:${mysql_connector_java_version}"
+
+    implementation "org.zalando:problem-spring-web:${zalando_problem_version}"
+    implementation "com.ibm.icu:icu4j:${ibm_icu4j_version}"
+    implementation "com.github.seancfoley:ipaddress:${github_seancfoley_ipaddress_version}"
+    implementation "org.apache.maven:maven-model:${apache_model_version}"
+    implementation "org.apache.pdfbox:pdfbox:${apache_pdfbox_version}"
+
+    implementation "org.springframework.boot:spring-boot-starter-artemis:${spring_boot_version}"
+
+    annotationProcessor "org.hibernate:hibernate-jpamodelgen:${hibernate_version}"
+    annotationProcessor("org.glassfish.jaxb:jaxb-runtime:${jaxb_runtime_version}") {
+        exclude group: "javax.ws.rs", module: "jsr311-api"
+    }
+
+    developmentOnly "org.springframework.boot:spring-boot-devtools:${spring_boot_version}"
+    testImplementation "com.h2database:h2:${h2_database_version}"
+
+}
+
+task executeTests(type: Exec) {
+    commandLine "./gradlew", "test"
+}
+
+// Command to execute the JavaDoc checkstyle verification ./gradlew checkstyleMain
+checkstyle {
+    toolVersion "${checkstyle_version}"
+    configFile file("../checkstyle.xml")
+    checkstyleTest.enabled = false
+    maxErrors = 0
+}
+
+nohttp {
+    source.include "build.gradle", "README.md"
+}
+
+task executeCheckstyle(type: Exec) {
+    commandLine "./gradlew", "checkstyleMain"
```

```suggestion
    commandLine "./gradlew", ":lecture:checkstyleMain"
```
Probably the same as with the `test` command above.